core/src/main/java/software/amazon/awssdk/pagination/Paginated.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2010-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.pagination;
+
+/**
+ * Return type interface for the paginated operations
+ *
+ * @param <PageT> A single response page
+ * @param <ItemT> The paginated item in a single page
+ */
+public interface Paginated<PageT, ItemT> extends SdkIterable<PageT> {
+
+    PageT firstPage();
```

Comment:
If we are going with generating both request/response and paginated methods I question the usefulness of this.

Reply:
Customers can use this as return type instead of operation specific class.
For example,
Paginated<ListTablesResponse, String> response = dbClient.listTables();
Instead of  
ListTablesPaginator response = dbClient.listTables();

Reply:
Sorry I meant the firstPage specifically. That was added mainly when we were thinking of only exposing the paginator but if that's no longer true I don't think it holds it's weight.

Reply:
We are eagerly loading the responses. So if a customer want a single page and uses next() method, SDK makes two service calls. This method might help in that case.

Reply:
If they just want a single page they can use the method that returns a single page. The only reason this was added was because we weren't planning on exposing the low level API method. Also I'm a little unsure about making service calls the customer didn't ask for, can we make it lazy loaded? You may still have to preemptively fetch the next page if hasNext is called (and you are at the end of a page boundary).

Reply:
> You may still have to preemptively fetch the next page if hasNext is called (and you are at the end of a page boundary).

We don't want to make service calls in hasNext(). It can be called multiple times before next() method is called. I didn't get the statement about end of page boundary.